presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1187,11 +1187,11 @@ public OptionalLong metadataDelete(ConnectorSession session, ConnectorTableHandl
             throw new TableNotFoundException(handle.getSchemaTableName());
         }
 
+        if (!table.get().getTableType().equals(TableType.MANAGED_TABLE.toString())) {
```

Comment:
Please add a test that queries fails with proper message when a `DELETE` or `INSERT` is issued on external tables. It could be either unit or product test.